### PR TITLE
Fix docker-compose-check Hook

### LINKED ISSUE #108: Fix docker-compose-check
# 🐛 Summary #

docker-compose-check is a pre-commit hook. It passes locally but fails in GitHub Actions with the error below.
Screenshot 2024-03-19 at 14 51 33

### PR DESCRIPTION
**Pull request by Matthew-Grayson**
#  #

## 🗣 Description ##
Add .env setup step to build.yml: addresses error by docker-compose-check hook.

## 💭 Motivation and context ##
The docker-compose-check hook relies on variables stored in a .env file. This file wasn't available in the GitHub Actions test environment, so I added a step to create it in the lint job.
Closes issue #108 

## 🧪 Testing ##
All unit tests pass.
docker-compose-check hook passes.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] *All* future TODOs are captured in issues, which are referenced
      in code comments.
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [ ] Tests have been added and/or modified to cover the changes in this PR.
- [ ] All new and existing tests pass.

## ✅ Pre-merge checklist ##

- [ ] Revert dependencies to default branches.
- [ ] Finalize version.

## ✅ Post-merge checklist ##

- [ ] Create a release.

### CODE REVIEW

**Review by cduhn17 — APPROVED** [review]
LGTM

**Comment by dav3r** [comment]
I'm confused when you say:
> Merging this and PR https://github.com/cisagov/XFD/issues/48 fixes the only remaining failing pre-commit hooks.

First of all, #48 is an issue, not a PR.  Did you mean PR #49, which resolved issue #48?

Second, this PR still shows those two JSON-related pre-commit hooks failing.  If those JSON hooks were passing thanks to the changes in #49, why are those changes not included in this PR?  If they were, then the `build / lint` tests in this PR should pass.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -100,7 +100,9 @@ jobs:
           python -m pip install --upgrade pip setuptools wheel
           pip install --upgrade --requirement requirements-test.txt
       - name: Set up pre-commit hook environments
-        run: pre-commit install-hooks
+        run: |
+          pre-commit install-hooks
+          cp dev.env.example .env
```
Comment (dav3r) [inline .github/workflows/build.yml]: I'd prefer to see this as two separate steps in the build workflow to more clearly explain what is going on:

- `name: Copy example .env file needed by docker-compose-check pre-commit hook` 
(feel free to adjust wording above as you see fit)
- `name: Set up pre-commit hook environments`

Comment (Matthew-Grayson) [reply .github/workflows/build.yml]: Done. https://github.com/cisagov/XFD/pull/111/commits/6f9c72705caefd92c0cc445fb0abeca45305d96c

Comment (dav3r) [reply .github/workflows/build.yml]: Can the step name ("Create .env file") please be updated to make it clearer that the .env file is required in order for the docker-compose-check pre-commit hook to pass?

**Comment by Matthew-Grayson** [comment]
> I'm confused when you say:
> 
> > Merging this and PR #48 fixes the only remaining failing pre-commit hooks.
> 
> First of all, #48 is an issue, not a PR. Did you mean PR #49, which resolved issue #48?
> 
> Second, this PR still shows those two JSON-related pre-commit hooks failing. If those JSON hooks were passing thanks to the changes in #49, why are those changes not included in this PR? If they were, then the `build / lint` tests in this PR should pass.

I've since merged develop into this branch and updated the PR language to reflect that all pre-commit hooks pass.

File: .github/workflows/build.yml (hunk 1 above)
Comment (Matthew-Grayson) [reply .github/workflows/build.yml]: Done. https://github.com/cisagov/XFD/pull/111/commits/7dc434bccd7090e215c480e30c909a7751856923

Comment (dav3r) [reply .github/workflows/build.yml]: Great, thanks!

**Review by dav3r — APPROVED** [review]
👍 👍

**Pull request merged**